src/lib/updateGrantsRecipients.js

```diff
@@ -387,25 +395,49 @@ export async function processFiles(hashSumHex) {
           && grantIds.includes(parseInt(g.replacement_grant_award_id, 10)),
       );
 
-      const grantUpdatePromises = grantsToUpdate.map((g) => (
-        Grant.unscoped().update(
-          { oldGrantId: parseInt(g.replaced_grant_award_id, 10) },
-          {
-            where: { id: parseInt(g.replacement_grant_award_id, 10) },
-            fields: ['oldGrantId'],
-            sideEffects: false,
-            transaction,
-            individualHooks: true,
+      const grantReplacementPromises = grantsToUpdate.map(async (g) => {
+        let grantReplacementType = await GrantReplacementTypes.findOne({
+          where: {
+            name: g.grant_replacement_type,
+          },
+        });
+
+        if (!grantReplacementType) {
+          // Create the new type
+          grantReplacementType = await GrantReplacementTypes.create({
+            name: g.grant_replacement_type,
+          });
+        }
+
+        const [grantReplacement, created] = await GrantReplacements.findOrCreate({
```

Comment:
we've been removing instances of findOrCreate from our codebase since (I believe, @GarrettEHill might know better) the creates run outside of transactions.

Reply:
yes please do not use findOrCreate its a mess

Reply:
Yea I would double check this with Garrett.